Webreidi hackathon2022 (#342)
NuevoFoundation/workshops: content/english/python-basics/conditional-statements-loops/nested-loops.md

### Nested loops

> *Nested loops*, builds off what we learned about for-loops and while loops in the previous exercise. You might want to review <a href="../../../python-basics/loops" target="blank">For-loop and While-loop</a> before we go on.

Reviewer: ```suggestion
*Nested loops*, builds off what we learned about for-loops and while loops in the previous exercise. You might want to review the <a href="../../../python-basics/loops" target="blank">For-loop and While-loop</a> page before we move on.
```

     y = 0
 ```
This is the output of our example.

 ```output
Go through outer while loop, x is 0 y is 0
Go through inner while loop, x is 0 y is 0
Go through inner while loop, x is 0 y is 1
Go through inner while loop, x is 0 y is 2
Go through inner while loop, x is 0 y is 3
Go through inner while loop, x is 0 y is 4
Go through outer while loop, x is 1 y is 0
Go through inner while loop, x is 1 y is 0
Go through inner while loop, x is 1 y is 1
Go through inner while loop, x is 1 y is 2
Go through inner while loop, x is 1 y is 3
Go through inner while loop, x is 1 y is 4
Go through outer while loop, x is 2 y is 0
Go through inner while loop, x is 2 y is 0
Go through inner while loop, x is 2 y is 1
Go through inner while loop, x is 2 y is 2
Go through inner while loop, x is 2 y is 3
Go through inner while loop, x is 2 y is 4
 ```

> #### <i>Questions</i>

Reviewer: ```suggestion
#### Questions
```

Can you try to find out how x and y values change and explain why?
<br/>
(<b>Hint</b>: How many times do we go through the outer loop and inner loop separately?)

<hr/>

> ### <b>Challenge One</b>

Reviewer: ```suggestion
### Challenge One
```

    print("")
```
This is the output of our example.

```output
( 0 , 0 ) ( 0 , 1 ) ( 0 , 2 ) 
( 1 , 0 ) ( 1 , 1 ) ( 1 , 2 ) 
( 2 , 0 ) ( 2 , 1 ) ( 2 , 2 ) 
```

> #### <i>Questions</i>

Reviewer: ```suggestion
#### Questions
```

Can you try to explain the printing pattern?
<br/>
(<b>Hint</b>: How many times do we go through the outer loop and inner loop separately?)

<hr/>

> ### <b>Challenge Two</b>

Reviewer: ```suggestion
### Challenge Two
```

Try to change the integers in the nested for loop and predict the outputs.
Then run the code and observe the outputs. Do they match your predictions?

<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/Basic-nested-for?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

#### <i>Questions</i>
Try to initialize n and m. Will this change our output from the nested loops and why?
Then run the code and observe the outputs. Do they match your predictions?

<iframe height="600px" width="100%" src="https://repl.it/@nuevofoundation/Nested-for-ex2?lite=true" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

> #### <i>Questions</i>

Reviewer: ```suggestion
#### Questions
```

<i>2</i>. What should we do if we want start printing from (1, 1)?
<br/>

> (<b>Hint</b>: What would happen if we change the ranges?)

Reviewer: Let's change `<i>` to use a single *
and `<b>` to use two *

> ### Challange Three ###

Reviewer: ```suggestion
### Challenge Three
```
